File: video-series-mono/src/main/java/org/jesperancinha/video/data/VideoSeriesDto.java
package org.jesperancinha.video.data;

import lombok.Builder;

import java.math.BigDecimal;

public record VideoSeriesDto(
        Long id,
        String name,
        Integer volumes,
        BigDecimal cashValue,
        Genre genre
) {
    @Builder
    public VideoSeriesDto {}
}

File: video-series-mono/src/main/kotlin/org/jesperancinha/video/data/VideoSeriesDto.kt
package org.jesperancinha.video.data

import com.fasterxml.jackson.annotation.JsonProperty
import java.math.BigDecimal

data class VideoSeriesDto (
    @param:JsonProperty("id")
    val id: Long,
    @param:JsonProperty("name")
    val name: String,
    @param:JsonProperty("volumes")
    val volumes: Int = 1,
    @param:JsonProperty("cashValue")
    val cashValue: BigDecimal,
    @param:JsonProperty("genre")
    val genre: Genre
)